query-connector/src/app/queryBuilding/buildFromTemplates/BuildFromTemplates.tsx

```diff
@@ -119,13 +119,11 @@ const BuildFromTemplates: React.FC<BuildFromTemplatesProps> = ({
         return;
       }
       const result = await getSavedQueryById(selectedQuery.queryId);
-
       if (result === undefined) {
         return; // todo: error???
       }
-
       const initialState: NestedQuery = {};
-      const savedQuery = result[0]; // what is this shape, why first index
```

Comment:
This is a result of the weird way we're typing `getSavedQueryById` to return an array of results rather than a single one. Will take care of this in a followup PR!